sdk/nodejs/provider.ts

```diff
@@ -2223,9 +2223,95 @@ export namespace yaml {
         const kind = obj["kind"];
         const apiVersion = obj["apiVersion"];
 
-        // Flatten `v1.List`. `v1.List` is an undocumented Kubernetes resource, and does not appear
-        // in the Kubernetes OpenAPI spec. `kubectl`, ksonnet, and Helm all flatten it.
-        if (apiVersion == "v1" && kind == "List") {
+        // Recursively traverse built-in Kubernetes list types into a single set of "naked" resource
+        // definitions that we can register with the Pulumi engine.
+        //
+        // Kubernetes does not instantiate list types like `v1.List`. When the API server receives
+        // a list, it will recursively traverse it and perform the necessary operations on the
+        // each "instantiable" resource it finds. For example, `kubectl apply` on a
+        // `v1.ConfigMapList` will cause the API server to traverse the list, and `apply` each
+        // `v1.ConfigMap` it finds.
+        //
+        // Since Kubernetes does not instantiate list types directly, Pulumi also traverses lists
+        // for resource definitions that can be managed by Kubernetes, and registers those with the
+        // engine instead.
+        if (
```

Comment:
How was this list generated?

What's the process for keeping it updated with newer k8s versions?

Reply:
It's a mustache template, right? It's generated here: https://github.com/pulumi/pulumi-kubernetes/pull/294/files#diff-551fbe30151ad564ca8ed6b6c8f46e40R98

That's picked up from the OpenAPI spec we provide. So it's always automatically up to date.